torchaudio/datasets/vctk.py

```diff
@@ -136,3 +140,99 @@ def __getitem__(self, n: int) -> Tuple[Tensor, int, str, str, str]:
 
     def __len__(self) -> int:
         return len(self._walker)
+
+
+class VCTK_092(Dataset):
+    """
+    Create a Dataset for VCTK 0.92, the latest version of the VCTK dataset.
+    Each item is a tuple of the form: (waveform, sample_rate, utterance, speaker_id, utterance_id)
+    Folder `p315` will be ignored due to the non-existent corresponding text files.
+    For more information about the dataset visit: https://datashare.is.ed.ac.uk/handle/10283/3443
+    """
+
+    _folder_txt = "txt"
+    _folder_audio = "wav48_silence_trimmed"
+    _ext_txt = ".txt"
+    _ext_audio = ".flac"
+    _mic_1 = "mic1"
+    _mic_2 = "mic2"
+
+    def __init__(self,
+                 root: str,
+                 url: str = URL,
+                 download: bool = False) -> None:
+
+        # Custom url may contain options like '?sequence=2&isAllowed=y'
+        # Slicing it off to extract only 'VCTK-Corpus-0.92.zip'
+        archive = os.path.join(root, os.path.basename(url)[:20])
+
+        self._path = os.path.join(root, FOLDER_IN_ARCHIVE_092)
+        self._txt_dir = os.path.join(self._path, self._folder_txt)
+        self._audio_dir = os.path.join(self._path, self._folder_audio)
+
+        if download:
+            if not os.path.isdir(self._path):
+                if not os.path.isfile(archive):
+                    checksum = _CHECKSUMS.get(url, None)
+                    download_url(url, root, hash_value=checksum, hash_type="md5")
+                extract_archive(archive, self._path)
+
+        if not os.path.isdir(self._path):
+            raise RuntimeError(
+                "Dataset not found. Please use `download=True` to download it."
+            )
+
+        # Extracting speaker IDs from the folder structure
+        self._speaker_ids = sorted(d for d in os.listdir(self._txt_dir)
+                                   if os.path.isdir(os.path.join(self._txt_dir, d)))
+        self._audio_ids = []
```

Comment:
I think storing `[(speaker_id, utterance_id), ...]` is enough.

If the dataset instance has `mic_id` attribute, then we can build the necessary path on With both `(speaker_id, utterance_id)`.

I think `load_sample(self, speaker_id, utterance_id, mic_id)` will give a more intuitive UX.

Reply:
Explain it again pls I didn't get it.

Reply:
What I meant is that currently, `self.sample_ids` is a list of string `<SPEAKER>_<UTTERANCE_KEY>_<MIC>` (here, I separate speaker and utterance for the purpose of clarity. I am aware that the variable `utterance_id` in the implementation includes `speaker_id` too), but now that microphone ID can be stored as `self.mic_id = mic_id`, we do not need to store `mic_id` inside of `self.sample_ids`.

Plus, rather than concatenating variables into one string (as `f{utterance_id}_{mic_id}` in the code) then splitting it again in `load_sample` method, storing the sample IDs as a tuple of `(<SPEAKER>, <UTTERANCE>)`, and passing it alongside with `mic_id`, to `load_sample` method will give even better separation of logic, in my opinion.

so what I am suggesting is something like

In `__init__`
```
    self.mic_id = mic_id
    ...
                 if speaker_id == 'p280' and mic_id == 'mic2':
                     break
                 if speaker_id == 'p362' and not os.path.isfile(audio_path_mic):  # <- Also I think flipping this if clause to use break will improve the readability of logic
                     break
                 self._sample_ids.append(utterance_id.split('_'))
```

In `__get_item__`
```
speaker_id, utterance_id = self.sample_ids[n]
self.load_item(speaker_id, utterance_id, self.mic_id)
```

This way the method signature of `load_item` express it better that 3 IDs are necessary and sufficient to locate one data sample from VCTK dataset.

Reply:
It will be `continue` in second `if` condition not `break` as we are not trying to skip the entire speaker if only one file doesn't exist. All else has been taken care of.